File: sub palindromes.java
/*input1: everest
output: 2
input2: abccbaab
output: 5
*/


import java.util.*;
public class Hello {

    public static void main(String[] args) {
		//Your Code Here
        Scanner sc=new Scanner(System.in);
        String c=sc.next();
        char[] s=c.toCharArray();
        int r=0,count;
        for(int i=0;i<s.length;i++){
            for(int j=i+1;j<s.length;j++){
                if(s[i]==s[j]){
                    count=0;
                    for(int s1=i,t=j;s1<=j&&t>=i;s1++,t--){
                        if(s[s1]==s[t])
                        count++;
                    }
                    if(count==(j-i)+1){
                        r++;
                    }
                }
            }
        }
        System.out.print(r);
	}
}

File: sub palindromes.py
/*input1: everest
output: 2
input2: abccbaab
output: 5
*/


def coun(s,n):
    d=[[0 for x in range(n)] for y in range(n)]
    p=[[False for x in range(n)] for y in range(n)]
    for i in range(n):
        p[i][i]=True
    for i in range(n-1):
        if(s[i]==s[i+1]):
            p[i][i+1]=True
            d[i][i+1]=n
    for g in range(2,n):
        for i in range(n-g):
            j=g+i
            if(s[i]==s[j] and p[i+1][j-1]):
                p[i][j]=True
            if(p[i][j]==True):
                d[i][j]=(d[i][j-1]+d[i+1][j]+1-d[i+1][j-1])
            else:
                d[i][j]=(d[i][j-1]+d[i+1][j]-d[i+1][j-1])
    return d[0][n-1]
s=input()
n=len(s)
print(coun(s,n))